activesupport/test/core_ext/hash_ext_test.rb

```diff
@@ -1523,9 +1523,9 @@ def test_should_use_default_proc_for_unknown_key
     assert_equal 3, hash_wia[:new_key]
   end
 
-  def test_should_use_default_proc_if_no_key_is_supplied
+  def test_should_return_nil_if_no_key_is_supplied
     hash_wia = HashWithIndifferentAccess.new { 1 +  2 }
-    assert_equal 3, hash_wia.default
+    assert_equal nil, hash_wia.default
```

Comment:
Yes, I guess this test was wrong. (added on #16356) .
```ruby
irb(main):001:0> h = Hash.new { 1 + 2}
=> {}
irb(main):002:0> h.default
=> nil
```

can you confirm that @senny ?